scripts/apps/archive/services/FamilyService.js

```diff
@@ -13,20 +31,150 @@ export function FamilyService(api, desks) {
             filter.push({not: {term: {unique_id: excludeItem.unique_id}}});
         }
 
-        return api('search').query({
+        return query(filter, 'versioncreated', 'desc');
+    };
+
+    /**
+     * @ngdoc method
+     * @name family#query
+     * @private
+     * @description Creates the query object and performs the query
+     * @param {Object} filter
+     * @param {string} sortField
+     * @param {string} order
+     * @param {Object} queryString
+     * @returns {Object}
+     */
+    const query = (filter, sortField, order, queryString) => {
+        let params = {
             repo: repo,
             source: {
                 query: {filtered: {filter: {
                     and: filter
                 }}},
-                sort: [{versioncreated: 'desc'}],
-                size: 100,
+                size: 200,
                 from: 0
             }
-        });
+        };
+
+        if (queryString) {
+            params.source.query.filtered.query = queryString;
+        }
+
+        params.source.sort = {};
+        params.source.sort[sortField] = order;
+        return api.query('search', params);
+    };
+
+    /**
+     * @ngdoc method
+     * @name family#fetchRelatedItems
+     * @public
+     * @description Returns takes, updates, corrections and kills of a given event_id
+     * @param {string} eventId
+     * @returns {Object}
+     */
+    this.fetchRelatedItems = (eventId) => {
+        var filter = [
+            {not: {term: {state: 'spiked'}}},
+            {term: {event_id: eventId}},
+            {not: {term: {type: 'composite'}}}
+        ];
+
+        return query(filter, 'versioncreated', 'asc');
+    };
+
+    /**
+     * @ngdoc method
+     * @name family#fetchRelatableItems
+     * @public
+     * @description Returns any story potentially linkable
+     * @param {string} keyword - slugline to be matched
+     * @param {string} sluglineMatch - type of matching rule for slugline
+     * @param {string} eventId - event id to be searched
+     * @param {string} modificationDateAfter - filter for versioncreated
+     * @returns {Object}
+     */
+    this.fetchRelatableItems = (keyword, sluglineMatch, eventId, modificationDateAfter) => {
+        var filter = [
```

Comment:
Do we need to add filter for the `last_published_version` for the items from the published collection?